Fix Clock for the old frontend

perost opened the pull request:
- Disable the dummy declaration for Clock in NFModelicaBuiltin and add
  it when building the top scope instead, since the old frontend
  incorrectly uses NFModelicaBuiltin in some cases.

casella commented:
@perost, you wrote "for the old frontend", but this PR had an [impact](https://libraries.openmodelica.org/branches/history/master/2021-12-03%2022:41:23..2021-12-04%2015:09:32.html) (both positive and negative) on other stuff too. 

Is this expected?

perost commented:
> @perost, you wrote "for the old frontend", but this PR had an [impact](https://libraries.openmodelica.org/branches/history/master/2021-12-03%2022:41:23..2021-12-04%2015:09:32.html) (both positive and negative) on other stuff too.
> 
> Is this expected?

Except for unstable models it only affected some ModelicaTest 3.2.1/3.2.2 and Modelica 3.1/3.2.1/3.2.2, but if you look at the same models in ModelicaTest 3.2.3 and Modelica 3.2.3 they behave the same way. So it was probably due to either some change behind the scenes or the models are unstable.

I also reverted that change later when I fixed it properly, and [it didn't affect anything](https://libraries.openmodelica.org/branches/history/master/2021-12-06%2011:23:14..2021-12-06%2022:32:35.html) (the Buildings improvements were due to my other commit in that report, and the regressions are just unstable models).

casella commented:
> Except for unstable models it only affected some ModelicaTest 3.2.1/3.2.2 and Modelica 3.1/3.2.1/3.2.2, but if you look at the same models in ModelicaTest 3.2.3 and Modelica 3.2.3 they behave the same way. So it was probably due to either some change behind the scenes or the models are unstable.

I guessed so, but I was wondering what this could have been due to. Also, I'm not sure I understand "unstable" in this context. For example, [TestWaterPumpDCMotorHeatTransfer](https://libraries.openmodelica.org/branches/master/ModelicaTest_3.2.1/files/ModelicaTest_3.2.1_ModelicaTest.Fluid.TestComponents.Machines.TestWaterPumpDCMotorHeatTransfer.sim) in ModelicaTest 3.2.1 fails because of
```
assert            | warning | The following assertion has been violated at time 0.000000
|                 | |       | pump.delta_head_init >= 0.0
assert            | warning | Variable violating min constraint: 0.0 <= pump.delta_head_init, has value: -50.1499
assert            | error   | Solving a linear system of equations with function
|                 | |       | "Matrices.solve" is not possible, because the system has either
|                 | |       | no or infinitely many solutions (A is singular).
assert            | error   | Failed to solve the initialization problem with global homotopy with equidistant step size.
assert            | debug   | Unable to solve initialization problem.
assert            | info    | simulation terminated by an assertion at initialization
```
If the system is found to be singular, it will always be found like that, unless we change the implementation of the linear algebra solver (which we didn't change recently, AFAIK). So that's pretty strange.

On the other hand, I don't really care much about old, pre-3.2.3 MSL stuff.

perost commented:
I opened #8293 for the redeclare issue and will try to fix it.

casella commented:
> Looking through the reports I found that the 3.2.3/4.0.0 versions of the same models failed in an [earlier report](https://libraries.openmodelica.org/branches/history/master/2021-11-30%2011:45:15..2021-11-30%2022:34:57.html) due to some redeclare fixes I made. Not sure why the 3.2.1/3.2.2 models only started to fail later, but it does seem to be the redeclare changes that are the actual culprit.

@sjoelund, perhaps older libraries are only checked every other time or something?

sjoelund commented:
MSL 3.2.1 and 3.2.2 are old libraries and only tested weekly as far as I remember

perost commented:
> MSL 3.2.1 and 3.2.2 are old libraries and only tested weekly as far as I remember

Are you sure? In that case [this report](https://libraries.openmodelica.org/branches/history/master/2021-12-03%2022:41:23..2021-12-04%2015:09:32.html) is completely wrong, since the regressions should really have shown up in [this report](https://libraries.openmodelica.org/branches/history/master/2021-11-30%2011:45:15..2021-11-30%2022:34:57.html).

sjoelund commented:
> Are you sure? In that case this report is completely wrong, since the regressions should really have shown up in this report.

It runs on Saturdays and only the one from a Saturday showed old MSL libraries. We could trigger old libraries manually if you want though.

perost commented:
> > Are you sure? In that case this report is completely wrong, since the regressions should really have shown up in this report.
> 
> It runs on Saturdays and only the one from a Saturday showed old MSL libraries. We could trigger old libraries manually if you want though.

The problem is not when it's run, it's that the listed commit does not have anything with the regressions to do. If we only do run those libraries once a week it should show all the commits that were involved.

sjoelund commented:
Yeah, that's probably problematic. And not easy to resolve since only some parts of it would use those commits...

perost commented:
> Yeah, that's probably problematic. And not easy to resolve since only some parts of it would use those commits...

Can't we run those libraries as a separate job?

sjoelund commented:
We could, but they are stored inside of the master table. The job generating the reports just look at the database as far as I remember. So it doesn't know which are new/old libraries

casella commented:
> The problem is not when it's run, it's that the listed commit does not have anything with the regressions to do. If we only do run those libraries once a week it should show all the commits that were involved.

Exactly.

I don't really see the point of running old library tests once a week. They are known to have issues, they are no longer maintained, and completely superseded by backwards-compatible maintained stuff. 

Additionally, any regression is reported in the wrong place, so not only it is hardly useless, it is in fact confusing.

@sjoelund, I would suggest to make that old library job manual. We trigger it on an individual basis if we have some good reasons to.
